### DISCUSSION TITLE
Start-Up Resources

### DISCUSSION THREAD

**Discussion by ahmutkadim** [godot-extended-libraries/godot-ideas#55] [Ideas]
A feature that I saw in some game engines is not present on Godot. This feature is very easy to implement and saves a lot of time which is why I am mentioning it.

When we wish to make a demo of our game, we would like to use simple, primitive graphics, like squares, triangles and circles. However these primitive resources are not present in Godot editor so we have to import them each time we need them, from an outside source. This can take time for especially in prototype phase.

What I ask is to implement this primitive resources like primitive sprite graphics and simple 3D materials directly to Godot Engine. For instance, when you view the "texture" property of a sprite2D node in the editor, there can be an option of "new square". And we can modify its color with modulation assuming that default square will have a color of white.

This would be great if implemented in core editor of the engine since such primitive resources can be used for many things, from placeholders to debug.

**Comment by quirkylemon** [comment]
I am interested in this. 
also simple 3d meshes are kinda already included
![image](https://github.com/godot-extended-libraries/godot-ideas/assets/43453305/7717f178-cba9-43b1-b0bf-4e82e3a6af02)

**Comment by Hephaistos847** [comment]
Doesn't Polygon2d kind of do this?
